schemas/code_signature.yml

```diff
@@ -0,0 +1,59 @@
+---
+- name: code_signature
+  title: Code Signature
+  group: 2
+  description: These fields contain information about binary code signatures.
+  type: group
+  reusable:
+    top_level: false
+    expected:
+      - file
+      - process
+      # - dll
+      # - driver
+  fields:
+
+    - name: subject_name
+      format: string
+      level: core
+      type: keyword
+      description: Subject name of the code signer
+      example: Microsoft Corporation
+
+
+    - name: valid
+      level: core
+      type: boolean
+      description: Boolean to capture if the digital signature is verified against the binary content.
+      example: "true"
+
+
+    - name: exists
+      format: core
+      level: extended
+      type: keyword
+      short: Boolean to capture if a signature is present.
+      description: >
+        Boolean to capture if a signature is present.
+
+        This should only populated if the signature was checked.
+      example: "true"
+
+    - name: trusted
+      format: core
+      level: extended
+      type: boolean
+      description: Stores the trust status of the certificate chain.
+      example: "true"
+
+
+    - name: status
+      format: string
+      level: extended
```

Comment:
Like for the field `exists`, please remove the `format:` attribute. And let's clarify which fields are core and extended. Here's an approach I think would make sense:

Core: can be determined via direct observation
Extended: need to evaluate the certificate validity to determine

If we go with this heuristic, the result would be:

Field | Level | Notes
--|--|--
`subject_name` | core | 
`valid` | extended | 
`exists` | core | Provided you agree with my comment on the description of the field
`trusted` | extended | 
`status ` | extended |

Reply:
i like your heuristic. resolved in 40aec6a4947136339c935855fa29c17107b09247